File: t001437.cpp
#include <stdio.h>
#include <math.h>
bool isPrime(unsigned long n) {
    unsigned long f = (unsigned long)sqrt(n);
    unsigned long i;
    for(i = 2; i < f + 1; i++)
    {
        if(n % i == 0)
        {
            return false;
        } 
    }
    return true;
}

int main()
{
    int T;
    int i;
    unsigned long N;
    scanf("%d", &T);
    for(i = 0; i < T; i++)
    {
        scanf("%d", &N);
        if(isPrime(N))
        {
            printf("YES\n");
        }
        else
        {
            printf("NO\n");
        }
        
    }

}

File: t001437.py
import math
def is_prime(val):
    if val < 2:
        return 0
    k = int(math.sqrt(val))
    for i in range(2, k + 1):
        if val % i == 0:
            return 0
    return 1

for T in range(int(input())):
    N = int(input())
    if is_prime(N):
        print("YES")
    else:
        print("NO")